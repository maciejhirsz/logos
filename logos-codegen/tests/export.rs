#![cfg(feature = "debug")]

use std::path::Path;

use insta::assert_snapshot;

#[rstest::rstest]
#[case("complex")]
fn test_export(#[case] case: &str) {
    let mut input_path = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/data/export");
    input_path.push(case);
    input_path.push("input.rs");

    let input = std::fs::read_to_string(input_path).expect("Unable to read input.rs");

    let _ = logos_codegen::generate(input.parse().expect("Unable to parse input.rs"));

    let generated_dot = std::fs::read_to_string(format!("{case}_export_tmp/{case}.dot"))
        .expect("Unable to read dot file");
    let generated_mermaid = std::fs::read_to_string(format!("{case}_export_tmp/{case}.mmd"))
        .expect("Unable to read mermaid file");

    assert_snapshot!(format!("{case}-dot"), generated_dot);
    assert_snapshot!(format!("{case}-mmd"), generated_mermaid);

    // cleanup
    let _ = std::fs::remove_dir_all(format!("{case}_export_tmp"));
}
