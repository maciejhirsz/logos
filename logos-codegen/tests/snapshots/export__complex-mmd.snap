---
source: logos-codegen/tests/export.rs
expression: generated_mermaid
---
flowchart TB
n0["State 0"]
style n0 stroke:#000000
n0-->|"a..=r|t..=z"|n1
n0-->|"s"|n2
n1["State 1"]
style n1 stroke:#00C853
n2["State 2"]
style n2 stroke:#00C853
n2-->|"t"|n3
n3["State 3"]
style n3 stroke:#000000
n3-->|"r"|n4
n4["State 4"]
style n4 stroke:#00C853
n4-->|"a..=t|v..=z"|n5
n4-->|"u"|n6
n5["State 5"]
style n5 stroke:#00C853
n5-->|"a..=z"|n5
n6["State 6"]
style n6 stroke:#00C853
n6-->|"a..=b|d..=z"|n5
n6-->|"c"|n7
n7["State 7"]
style n7 stroke:#00C853
n7-->|"a..=s|u..=z"|n5
n7-->|"t"|n8
n8["State 8"]
style n8 stroke:#00C853
n8-->|"a..=z"|n5
